core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -161,14 +161,27 @@ private void setCandidates(List<Endpoint> candidates) {
 
             // Remove old contexts when the newly created contexts are fully initialized to smoothly transition
             // to new endpoints.
-            contextGroup.whenInitialized().thenRun(() -> {
+            contextGroup.whenInitialized().handle((unused, cause) -> {
+                if (cause != null && !initialized) {
+                    if (logger.isWarnEnabled()) {
+                        logger.warn("Failed to initialize the first health check. " +
```

Comment:
Optional; I feel like this log can be more explanatory since it might end up being a false-positive log (due to eventual success from retries)

```suggestion
                        logger.warn("The first health check failed for all endpoints. " +
```

Reply:
Nice suggestion.